apis/Google.Cloud.Datastore.V1/Google.Cloud.Datastore.V1/AggregationQueryResults.cs

```diff
@@ -0,0 +1,96 @@
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Protobuf.Collections;
+using Google.Protobuf.WellKnownTypes;
+using static Google.Cloud.Datastore.V1.QueryResultBatch.Types;
+
+namespace Google.Cloud.Datastore.V1;
+
+/// <summary>
+/// Represents the result of an <see cref="AggregationQuery"/>.
+/// </summary>
+public sealed class AggregationQueryResults
+{
+    /// <summary>
+    /// The result of the aggregation functions, ex: `COUNT(*) AS  total_entities`.
+    /// The key is the [alias][google.datastore.v1.AggregationQuery.Aggregation.alias] assigned
+    /// to the aggregation function on input and the size of this map equals the number
+    /// of aggregation functions in the query.
+    /// </summary>
+    private readonly MapField<string, Value> _aggregateProperties = new MapField<string, Value>();
+
+    /// <summary>
+    /// The condition that caused the query to complete, indicating whether more
+    /// results are available beyond the query limit or end cursor, or whether
+    /// the data has been exhausted.
+    /// </summary>
+    public MoreResultsType MoreResults { get; }
+
+    /// <summary>
+    /// The time at which the snapshot was read.
+    /// </summary>
+    public Timestamp ReadTime { get; }
+
+    internal AggregationQueryResults(MoreResultsType moreResults, Timestamp readTime, MapField<string, Value> aggregateProperties)
+    {
+        MoreResults = moreResults;
+        ReadTime = readTime;
+        _aggregateProperties = aggregateProperties;
+    }
+
+    /// <summary>
+    /// Returns the result of the aggregation function that is assigned to the given alias.
+    /// Will be null if <see paramref="alias" /> was not specified in the original query.
+    /// </summary>
+    /// <param name="alias">The alias of the aggregation function that was provided in the aggregation query.
+    /// If not provided,  Datastore will pick a default name following the format
+    /// `property_&lt;incremental_id++>`. For example:
+    ///```text
+    ///```
+    /// AGGREGATE
+    ///   COUNT_UP_TO(1) AS count_up_to_1,
+    ///   COUNT_UP_TO(2),
+    ///   COUNT_UP_TO(3) AS count_up_to_3,
+    ///   COUNT_UP_TO(4)
+    /// OVER (
+    ///   ...
+    /// );
+    /// ```
+    ///
+    /// becomes:
+    /// ```text
+    /// ```
+    /// AGGREGATE
+    ///   COUNT_UP_TO(1) AS count_up_to_1,
+    ///   COUNT_UP_TO(2) AS property_1,
+    ///   COUNT_UP_TO(3) AS count_up_to_3,
+    ///   COUNT_UP_TO(4) AS property_2
+    /// OVER (
+    ///   ...
+    /// );</param>
+    /// <returns>The result of the aggregation function corresponding to the given alias.</returns>
+    public Value this[string alias] => _aggregateProperties[alias];
+
+    internal static AggregationQueryResults FromRunAggregationQueryResponse(RunAggregationQueryResponse response)
+    {
+        MapField<string, Value> aggregateProperties = new MapField<string, Value>();
```

Comment:
If we used a plain dictionary, I *suspect* this could just be `response.Batch.AggregationResults.SelectMany(result => result.AggregateProperties).ToDictionary(pair => pair.Key, pair => pair.Value);` (with appropriate formatting...) I think that would make the structure a little clearer.